### ISSUE TITLE
[High Priority] Hitting the AWS Cloudwatch Logs GetLogEvents Limit

### ISSUE THREAD

**Issue by mantzas** [e-travel/cloudwatchlogsbeat#1]
Hi,

the application hit's the limits of [Cloudwatch Logs](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html).

Due to this it is almost impossible to watch logs from AWS Console and moving the logs to cloudwatchlogsbeat resulted in a 1h delay of logs, which makes it impossible to monitor.

**Suggestion:**

AWS recommends to [subscribe to logs](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Subscriptions.html) and send it somewhere else for processing like Kinesis, Lambda etc.

There we could implement a Lambda to send the data wherever we want.

Thanks

**Comment by mantzas** [comment]
the logs from show this problem:

`2017-08-02T08:37:57+03:00 ERR /aws/lambda/smsGateway-production-DlrReceiver/2017/08/01/[$LATEST]d879cb48502d43e49530e52c21c23614 ThrottlingException: Rate exceeded
2017-08-02T08:37:59+03:00 ERR /aws/lambda/flightAlerts-production-FaAlertProcessor/2017/08/01/[$LATEST]691c505a874b4503b05d2e18ad424f5a ThrottlingException: Rate exceeded
2017-08-02T08:38:05+03:00 ERR /aws/lambda/LewebDatastore-production-GetAddonsData/2017/07/31/[$LATEST]5e944195ae7349fbb9b9de1895a9d8a1 ThrottlingException: Rate exceeded
2017-08-02T08:38:44+03:00 ERR /aws/lambda/smsGateway-production-DlrReceiver/2017/08/01/[$LATEST]70f5eefdbdc64aa0a9ec522a570983fc ThrottlingException: Rate exceeded
2017-08-02T08:38:47+03:00 ERR /aws/lambda/smsGateway-production-DlrReceiver/2017/08/02/[$LATEST]cafc312efd9644c291134f7065ae3144 ThrottlingException: Rate exceeded
2017-08-02T08:38:48+03:00 ERR /aws/lambda/tripNotifications-production-TripNotificationsAddOn/2017/08/01/[$LATEST]115d8ea8048f4ac6b2cb9813a557b636 ThrottlingException: Rate exceeded
2017-08-02T08:39:00+03:00 ERR /aws/lambda/flightAlerts-production-FaAlertProcessor/2017/08/02/[$LATEST]9f7b77c9fa834cafa7fb5dca794625fe ThrottlingException: Rate exceeded
2017-08-02T08:40:33+03:00 ERR /aws/lambda/flightAlerts-production-FaAlertProcessor/2017/08/01/[$LATEST]34389068292e4125b290f804616adf3c ThrottlingException: Rate exceeded
2017-08-02T08:40:35+03:00 ERR /aws/lambda/smsGateway-production-DlrReceiver/2017/08/01/[$LATEST]3dc19bb361c74e38bed6b5a8cddcf072 ThrottlingException: Rate exceeded
2017-08-02T08:41:05+03:00 ERR /aws/lambda/tripNotifications-production-TripNotificationsAddOn/2017/08/01/[$LATEST]0c36e9a340c443b4b8293a7ea2a8abe6 ThrottlingException: Rate exceeded`

**Comment by kkentzo** [comment]
Indeed, the beat is subject to cloudwatchlogs limitation wrt the AWS API requests. The suggestion of following a push model is valid but probably out of scope in the context of this beat.

Throttling errors can be mitigated (not eliminated) by tuning the beat's configuration. As a follow-up action to this issue, we have introduced two categories of streams (standard and hot) which can be separately configured in the beat's configuration file (445fd714c91fa4b74c8a848307434cdcea3c8d12). A relevant discussion of this issue has also been added in the project's README.

Thanks.
